### PR TITLE
React Part 3b: Saving duplicate results and result_metrics directories

### PR DESCRIPTION
**Pull request by adisingh50**
In order for the React application to work on anyone else's machine, I've added some lines of python code to save duplicate copies of the `results` and `result_metrics` directories within my React app source code (in order for the web app to read the json and txt files internally and display the correct information).

### CODE REVIEW

**Comment by adisingh50** [comment]
Since I am not all too familiar with dask and how it's used to call the save file functions in this project, please correct my syntax if needed.

File: gtsfm/multi_view_optimizer.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
 from gtsfm.bundle.bundle_adjustment import BundleAdjustmentOptimizer
 from gtsfm.data_association.data_assoc import DataAssociation
 
+# Paths to Save Output in React Folders.
+REACT_METRICS_PATH = "rtf_vis_tool/src/result_metrics"
```
Comment (johnwlambert) [inline gtsfm/multi_view_optimizer.py:28]: Thanks for the PR, @adisingh50.

this should be relative to the file's location:
```
REACT_METRICS_PATH = Path(__file__).resolve().parent.parent / "rtf_vis_tool" / "src" / "result_metrics"
```

File: environment_mac.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@ dependencies:
   - matplotlib>=3.3.3
   - networkx
   - numpy
+  - nodejs
```
Comment (johnwlambert) [inline environment_mac.yml:30]: didnt we add this before?

File: gtsfm/scene_optimizer.py [hunk 3]
```diff
@@ -33,6 +33,10 @@
 METRICS_PATH = "result_metrics"
 RESULTS_PATH = "results"
 
+# Paths to Save Output in React Folders.
+REACT_METRICS_PATH = "rtf_vis_tool/src/result_metrics"
```
Comment (johnwlambert) [inline gtsfm/scene_optimizer.py:37]: same note here as before about relative paths to file, then converted to absolute with .resolve()

File: gtsfm/multi_view_optimizer.py (hunk 1 above)
Comment (adisingh50) [reply gtsfm/multi_view_optimizer.py:28]: Got it. I'll make this edit for both React file paths.

File: environment_mac.yml (hunk 2 above)
Comment (adisingh50) [reply environment_mac.yml:30]: I added `nodejs` to `environment_linux.yml`, but I figured I needed to add the same dependency for `environment_mac.yml`. Let me know if you think this isn't needed.

Comment (johnwlambert) [reply environment_mac.yml:30]: thanks @adisingh50 . Can you add it for Windows, also?

**Comment by johnwlambert** [comment]
Thanks for the PR, Adi.

The Python CI is failing because of `flake8` errors:
```
                                                     1974    329    83%
Collecting flake8
  Downloading flake8-3.9.1-py2.py3-none-any.whl (73 kB)
Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in /usr/share/miniconda3/envs/gtsfm-v1/lib/python3.8/site-packages (from flake8) (0.6.1)
Collecting pycodestyle<2.8.0,>=2.7.0
  Downloading pycodestyle-2.7.0-py2.py3-none-any.whl (41 kB)
Collecting pyflakes<2.4.0,>=2.3.0
  Downloading pyflakes-2.3.1-py2.py3-none-any.whl (68 kB)
Installing collected packages: pyflakes, pycodestyle, flake8
Successfully installed flake8-3.9.1 pycodestyle-2.7.0 pyflakes-2.3.1
gtsfm/multi_view_optimizer.py:31:1: E302 expected 2 blank lines, found 1
gtsfm/scene_optimizer.py:184:121: E501 line too long (140 > 120 characters)
gtsfm/scene_optimizer.py:458:110: W292 no newline at end of file
Error: Process completed with exit code 1.

```

**Comment by ayushbaid** [comment]
LGTM!

**Review by johnwlambert — APPROVED** [review]

**Pull request merged**